chef/cookbooks/ceph/recipes/mon.rb

```diff
@@ -142,13 +149,14 @@
 end
 
 execute "Create Ceph client.admin key when ceph-mon is ready" do
-  command "ceph-create-keys -i #{node['hostname']}"
+  command "ceph-create-keys -i #{mon_name}"
   not_if { File.exists?("/etc/ceph/#{cluster}.client.admin.keyring") }
 end
 
 get_mon_addresses.each do |addr|
   execute "peer #{addr}" do
-    command "ceph --admin-daemon '/var/run/ceph/ceph-mon.#{node['hostname']}.asok' add_bootstrap_peer_hint #{addr}"
+    command "ceph --admin-daemon '/var/run/ceph/ceph-mon.#{mon_name}.asok' " \
```

Comment:
you're changing the asok path here but don't adjust all other references in ceph/libraries/default.rb and other places